### ISSUE TITLE
open-not-closed rule False Negative

### ISSUE THREAD

**Issue by dlukeomalley** [returntocorp/sgrep-rules#569]
https://semgrep.live/nJWG

```yaml
rules:
  - id: open-never-closed
    patterns:
      - pattern: $FILE = open(...)
      - pattern-not-inside: |
          $FILE = open(...)
          ...
          $FILE.close()
    message: "file object opened without corresponding close"
    languages: [python]
    severity: ERROR
```

Fails to detect `file.txt` below:

```py
f = open("file.txt")
f.read()
f = open("file2.txt")
f.read()
f.close()
```

**Comment by minusworld** [comment]
I added this rule https://github.com/returntocorp/semgrep-rules/pull/568 to satisfy this case; is this satisfactory?

**Comment by nbrahms** [comment]
Relates to #552

**Comment by nbrahms** [comment]
Seems to be addressed by https://github.com/returntocorp/semgrep-rules/pull/568
